crates/ethereum/payload/src/lib.rs

```diff
@@ -344,39 +297,19 @@ where
     // check if we have a better block
     if !is_better_payload(best_payload.as_ref(), total_fees) {
         // Release db
-        drop(evm);
-
+        drop(strategy);
```

Comment:
I think the strategy can be dropped even earlier to free mutable ref to db taken on strategy instantiation